pkgdb/src/flox-flake.cc

```diff
@@ -38,24 +38,36 @@ namespace flox {
 
 /* -------------------------------------------------------------------------- */
 
-FloxFlake::FloxFlake( const nix::ref<nix::EvalState> & state,
-                      const nix::FlakeRef &            ref )
-try : state( state ),
-  lockedFlake( nix::flake::lockFlake( *this->state, ref, defaultLockFlags ) )
-  {}
-catch ( const std::exception & err )
-  {
+nix::flake::LockedFlake
+lockFlake( nix::EvalState &              state,
+           const nix::FlakeRef &         ref,
+           const nix::flake::LockFlags & flags )
+{
+  try
+    {
+      return nix::flake::lockFlake( state, ref, flags );
+    }
+  catch ( const std::exception & err )
+    {
+      throw LockFlakeException( "failed to lock flake \"" + ref.to_string()
+                                  + "\"",
+                                nix::filterANSIEscapes( err.what(), true ) );
```

Comment:
Aren't we using single ticks now for strings?
Not necessarily blocking for this PR but we should also check the string returned by pkgdb against the style guide

```suggestion
      throw LockFlakeException( "failed to lock flake '" + ref.to_string()
                                  + '",
                                nix::filterANSIEscapes( err.what(), true ) );
```

Reply:
Fixed. This was from Alex's branch and I just missed that during copy/paste.